### DISCUSSION TITLE
How to set minimum length of an accepted barCode?

### DISCUSSION THREAD

**Discussion by edu1525** [journeyapps/zxing-android-embedded#600] [Q&A]
#### Description of the problem:
In Brazil we have a default format of barcodes used for bank slips, its an ITF barCode with 44 characters, the problem is that sometimes this library recognizes smaller barCodes, with only some digits of the original barCode, it generates a false scan to user, because the beep is reproduced and the activity is closed when a barCode is found. Is there a way to specify the min number of characters to be recognized as a valid read?

**Which library version are you using?**  4.1.0.

**Which phone/tablet are you using, and which Android version does it run?** Samsung Galaxy S7 

**Does the same happen on other devices or an emulator?** Yes, on other devices too

**Comment by manoellribeiro** [comment]
hey @edu1525, have you found some workaround to read brazilian barcodes?

**Comment by rkistner** [comment]
I'd recommend:
1. Disable any barcodes types other than the specific ones you're interested in.
2. Start with the continuous scanning example: https://github.com/journeyapps/zxing-android-embedded/blob/master/sample/src/main/java/example/zxing/ContinuousCaptureActivity.java
3. Only stop the scanning once you have found a barcode with the minimum length.
